src/schematools/loaders.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Union
+
+from more_ds.network.url import URL
+
+if TYPE_CHECKING:
+    from schematools.types import DatasetSchema
+
+
+class SchemaLoader:
+    """Base class for schema loaders."""
+
```

Comment:
Define a instance variable `schema_url: Union[Path, URL]` according to https://www.python.org/dev/peps/pep-0526/

Reply:
Good point, tx!